gdnative/multi-lod/src/lodmanager.cpp

```diff
@@ -271,58 +260,66 @@ void LODManager::stop_loop() {
     lod_objects_semaphore->free();
 }
 
-void LODManager::add_object(Node* obj) {
-    // TODO: Print the object name.
-    debug_level_print(1, "Adding a new object to the LOD Manager's list.");
-
+void LODManager::add_object(LODBaseComponent* lbc) {
     // Go through array list to find one that has space
     if (use_multithreading) {
         lod_objects_semaphore->wait();
     }
 
+    debug_level_print(1, lbc->get_object()->get_name() + String(": Adding a new object to the LOD Manager's list"));
+
     bool added = false; // Keep track in case we need to add a new list
     for (int i = 0; i < lod_object_arrays.size(); i++) {
-        Array lod_objects = lod_object_arrays[i];
+        std::vector<LODBaseComponent*>& lod_objects = lod_object_arrays[i];
         if (lod_objects.size() < MAX_ARRAY_SIZE) {
-            lod_objects.push_back(obj);
+            lod_objects.push_back(lbc);
             lod_object_count++;
             added = true;
             break;
         }
     }
 
     if (!added) {
-        Array newLODObjectArray;
-        newLODObjectArray.push_back(obj);
+        std::vector<LODBaseComponent*> newLODObjectArray;
+        newLODObjectArray.push_back(lbc);
         lod_object_arrays.push_back(newLODObjectArray);
+        lod_object_count++; // why wasn't this here?
```

Comment:
Is it correct to add this? It looked like an oversight.

Reply:
Yes, good catch